src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -184,9 +184,12 @@ public override Iterator<TSource> Clone()
 
             public override void Dispose()
             {
-                if (_enumerator is IDisposable) ((IDisposable)_enumerator).Dispose();
-                _enumerator = null;
-                base.Dispose();
+                if (_enumerator != null)
+                {
+                    ((IDisposable)_enumerator).Dispose();
+                    _enumerator = null;
+                    base.Dispose();
```

Comment:
@stephentoub Fixed this in the latest commit; the cast is no longer there, and `base.Dispose()` has moved outside the block. I was keeping the cast before because I had seen it in other places throughout the BCL and there seem to be [a few](https://www.google.com/search?q=explicit+cast+to+idisposable) questions specifically about this, but I wasn't able to find any problems.